packages/iota/src/validation/outputs/common.ts

```diff
@@ -0,0 +1,347 @@
+    .set(3, BASIC_OUTPUT_MIN_UNLOCK_CONDITIONS)
+    .set(4, ALIAS_OUTPUT_MIN_UNLOCK_CONDITIONS)
+    .set(5, FOUNDRY_OUTPUT_MIN_UNLOCK_CONDITIONS)
+    .set(6, NFT_OUTPUT_MIN_UNLOCK_CONDITIONS);
+
+/**
+ * Unlock Condition type names.
+ */
+const ULOCK_CONDITION_TYPE_NAMES = [
+    "Address Unlock Condition", 
+    "Storage Deposit Return Unlock Condition",
+    "Timelock Unlock Condition",
+    "Expiration Unlock Condition",
+    "State Controller Address Unlock Condition",
+    "Governor Address Unlock Condition",
+    "Immutable Alias Address Unlock Condition"
+];
+
+/**
+ * Allowed Basic output features.
+ */
+const BASIC_OUTPUT_FEATURES = [
+    SENDER_FEATURE_TYPE,
+    METADATA_FEATURE_TYPE,
+    TAG_FEATURE_TYPE
+];
+
+/**
+ * Allowed Alias output features.
+ */
+const ALIAS_OUTPUT_FEATURES = [
+    SENDER_FEATURE_TYPE,
+    METADATA_FEATURE_TYPE
+];
+
+/**
+ * Allowed Alias output immutable features.
+ */
+const ALIAS_OUTPUT_IMMUTABLE_FEATURES = [
+    ISSUER_FEATURE_TYPE,
+    METADATA_FEATURE_TYPE
+];
+
+/**
+ * Allowed Foundry output features.
+ */
+const FOUNDRY_OUTPUT_FEATURES = [METADATA_FEATURE_TYPE];
+
+/**
+ * Allowed Foundry output immutable features.
+ */
+const FOUNDRY_OUTPUT_IMMUTABLE_FEATURES = FOUNDRY_OUTPUT_FEATURES;
+
+/**
+ * Allowed Nft output features.
+ */
+ const NFT_OUTPUT_FEATURES = BASIC_OUTPUT_FEATURES;
+
+ /**
+ * Allowed Nft output immutable features.
+ */
+ const NFT_OUTPUT_IMMUTABLE_FEATURES = ALIAS_OUTPUT_IMMUTABLE_FEATURES;
+
+/**
+ * Map Output type to allowed features.
+ */
+const OUTPUT_TYPE_TO_FEATURES = new Map()
+    .set(3, BASIC_OUTPUT_FEATURES)
+    .set(4, ALIAS_OUTPUT_FEATURES)
+    .set(5, FOUNDRY_OUTPUT_FEATURES)
+    .set(6, NFT_OUTPUT_FEATURES);
+/**
+ * Map Output type to allowed immutable features.
+ */
+const OUTPUT_TYPE_TO_IMMUTABLE_FEATURES = new Map()
+    .set(4, ALIAS_OUTPUT_IMMUTABLE_FEATURES)
+    .set(5, FOUNDRY_OUTPUT_IMMUTABLE_FEATURES)
+    .set(6, NFT_OUTPUT_IMMUTABLE_FEATURES);
+
+/**
+ * Unlock Condition type names.
+ */
+const FEATURE_TYPE_NAMES = [
+    "Sender Feature", 
+    "Issuer Feature",
+    "Metadata Feature",
+    "Tag Feature"
+];
+
+/**
+ * Map Output type to name.
+ */
+const OUTPUT_TYPE_NAMES = new Map()
+    .set(3, "Basic")
+    .set(4, "Alias")
+    .set(5, "Foundry")
+    .set(6, "Nft");
+
+/**
+ * Validate the output common rules.
+ * @param output The output to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateCommonRules(output: OutputTypes, protocolInfo: INodeInfoProtocol): IValidationResult {
+    const results: IValidationResult[] = [];
+    const outputType = output.type;
+    const outputName = OUTPUT_TYPE_NAMES.get(output.type);
+
+    results.push(validateAmountIsGreaterThanZero(output.amount, outputName));
+
+    results.push(validateAmountIsLesserThanMaxSupply(output.amount, protocolInfo.tokenSupply, outputName));
+    
+    if (outputType !== TREASURY_OUTPUT_TYPE) {
```

Comment:
I the type of `output` param is more specific, we can get away with not having this check.
i.e.
```
export function validateCommonRules(output: IBasicOutput | IAliasOutput | IFoundryOutput | INftOutput ...
```
We can also create an "alias" somewhere above for this type.
`type supportedOutputTypes = IBasicOutput | IAliasOutput | IFoundryOutput | INftOutput`
or
spread the parameters to one per line so it's easily readable.